Content/Items/Misc/Weapons.GeodeBow.cs

```diff
@@ -0,0 +1,482 @@
+
+            for (int j = 0; j < 16; j++)
+            {
+                Vector2 direction = Main.rand.NextVector2CircularEdge(1, 1);
+                Projectile shrapnel = Projectile.NewProjectileDirect(Projectile.GetSource_FromThis(), Projectile.Center + (direction * 10), direction * Main.rand.Next(5, 10) * (float)Math.Sqrt(counter), ModContent.ProjectileType<GeodeBowShrapnel>(), 0, 0, owner.whoAmI);
+                (shrapnel.ModProjectile as GeodeBowShrapnel).width = Main.rand.Next(30, 50);
+                shrapnel.timeLeft = Main.rand.Next(90);
+            }
+
+            Vector2 vel = Vector2.Normalize(offset) * 5;
+            for (int i = 1; i <= 4; i++)
+                Gore.NewGore(Projectile.GetSource_FromThis(), Projectile.Center, vel.RotatedByRandom(0.4f), Mod.Find<ModGore>("GeodeBowGrowthGore" + i.ToString()).Type, Projectile.scale);
+            Projectile.active = false;
+        }
+    }
+
+    internal class GeodeBowExplosion : ModProjectile
+    {
+        public override string Texture => AssetDirectory.Invisible;
+
+        public float radiusMult => Projectile.ai[1];
+
+        public float Progress => 1 - (Projectile.timeLeft / 10f);
+
+        private float Radius => (150 + (15 * Projectile.ai[0])) * (float)(Math.Sqrt(Progress)) * radiusMult;
+
+        public override void SetDefaults()
+        {
+            Projectile.width = 80;
+            Projectile.height = 80;
+            Projectile.DamageType = DamageClass.Ranged;
+            Projectile.friendly = true;
+            Projectile.tileCollide = false;
+            Projectile.penetrate = -1;
+            Projectile.timeLeft = 10;
+        }
+
+        public override void SetStaticDefaults()
+        {
+            DisplayName.SetDefault("Geode Explosion");
+        }
+
+        public override void AI()
+        {
+            var crystals = Main.projectile.Where(x => x.active && x.type == ModContent.ProjectileType<GeodeBowGrowth>()).ToList();
+
+            foreach (Projectile proj in crystals)
+            {
+                Vector2 line = proj.Center - Projectile.Center;
+                line.Normalize();
+                line *= Radius;
+
+                if (!Collision.CheckAABBvLineCollision(proj.position, proj.Size, Projectile.Center, Projectile.Center + line))
+                    continue;
+
+                Vector2 vel = Vector2.Normalize((proj.ModProjectile as GeodeBowGrowth).offset) * 5;
+                for (int i = 1; i <= 4; i++)
+                    Gore.NewGore(Projectile.GetSource_FromThis(), proj.Center, vel.RotatedByRandom(0.4f), Mod.Find<ModGore>("GeodeBowGrowthGore" + i.ToString()).Type, proj.scale);
+                proj.active = false;
+            }
+        }
+
+        public override bool PreDraw(ref Color lightColor) => false;
```

Comment:
dont use arrow notation in the middle of a class like this

Reply:
Fixed